### ISSUE TITLE
Binary classification just learns the major class

### ISSUE THREAD

**Issue by ro6842** [tflearn/tflearn#454]
I am using TFlearn to train a case about binary classifiction, expected output is 0 or 1.
And the training Input is images' pixels corresponding to labels 0 or 1, but the major labels are 0.
For example, Input is 60*80 = 4800 pixels and the number of label 0 has 3800, the number of labe 1 has 1000. In this situation, I start to training.

First iteration, the accuracy is about 0.73
...
Until 200 iteration when it almost converges, the accuracy is about 0.79

And I use model.predict to predict the model, I found the model only learn the label 0 class, the label 1 class just be considered the wrong case. When the iteration increases, the label 0 class learn better, but the label 1 class can't learn at all.

For example, Using model.predict to predict 0 class gets [[0.8630272746086121, 0.02531861700117588]]
                     Using model.predict to predict 1 class gets [[0.8631386756896973,0.025179509073495865]]

It almost the same!! But I want the TFlearn can learn 0 and 1 class better and better when iteration increases.

Whether I am doing wrong, or my network structure has to improve?
Below is my network structure:

		tnorm = tflearn.initializations.truncated_normal(mean=0.0, stddev=0.01)
		network = input_data(shape=[None, 27, 27, 1])
		network = conv_2d(network, 6, filter_size = 5, activation='relu')
		network = max_pool_2d(network, 3)
		network = conv_2d(network, 16, filter_size = 5, activation='relu')
		network = max_pool_2d(network, 3)
		network = fully_connected(network, 120, activation='sigmoid', weights_init=tnorm)
		network = fully_connected(network, 2, activation='sigmoid', weights_init=tnorm)
		network = regression(network, optimizer='sgd',loss='binary_crossentropy',learning_rate=0.001, n_classes = 2)
			
		model = tflearn.DNN(network, tensorboard_verbose=0)
		model.fit(X, Y, n_epoch=10, shuffle=True, validation_set=(X_test, Y_test),show_metric=True, batch_size=100, run_id='bgsub_cnn')

Hopefully you can understand my question, I can explain more if necessary.

**Comment by aymericdamien** [comment]
Yes, you can simply use np.argmax(....)

**Comment by robben1234** [comment]
[[0.0, 1.0, 0.0 ... 0.0]] is a bit confusing because you expect it to be array of probabilities (and it's sum to be equal 1) like [[0.3 0.1 0.6 0. 0. ... 0.]], but network is always answering with one index = 1 and every other = 0. Even when it's two object on picture.

**Comment by ro6842** [comment]
Thank you, I got it!

**Comment by ro6842** [comment]
Thank you, I got it!
